starlite/utils/serialization.py

```diff
@@ -20,4 +23,87 @@ def default_serializer(value: Any) -> Any:
         return value.get_secret_value()
     if isinstance(value, (PurePath, PurePosixPath)):
         return str(value)
-    raise TypeError("unsupported type")
+    if isinstance(value, UUID):
+        return str(value)
+    if isinstance(value, (datetime, date, time)):
+        return value.isoformat()
+    raise TypeError(f"Unsupported type: {type(value)!r}")
+
+
+def dec_hook(type_: Any, value: Any) -> Any:
+    """Transform values non-natively supported by `msgspec`
+
+    Args:
+        type_: Encountered type
+        value: Value to coerce
+
+    Returns:
+        A `msgspec`-supported type
+    """
+    if issubclass(type_, BaseModel):
+        return type_(**value)
+    raise TypeError(f"Unsupported type: {type(value)!r}")
+
+
+_msgspec_json_encoder = msgspec.json.Encoder(enc_hook=default_serializer)
+_msgspec_json_decoder = msgspec.json.Decoder(dec_hook=dec_hook)
+_msgspec_msgpack_encoder = msgspec.msgpack.Encoder(enc_hook=default_serializer)
+_msgspec_msgpack_decoder = msgspec.msgpack.Decoder(dec_hook=dec_hook)
```

Comment:
A `dec_hook` is only called for [typed decoding](https://jcristharif.com/msgspec/usage.html#typed-decoding), where an output `type` is specified. Since these decoders never specify a type, the `dec_hook` here is doing nothing and can be removed (perhaps I should add a warning for this in `msgspec`).

```suggestion
_msgspec_json_encoder = msgspec.json.Encoder(enc_hook=default_serializer)
_msgspec_json_decoder = msgspec.json.Decoder()
_msgspec_msgpack_encoder = msgspec.msgpack.Encoder(enc_hook=default_serializer)
_msgspec_msgpack_decoder = msgspec.msgpack.Decoder()
```

Reply:
There is actually a reason to have those in there. We want to support using `Structs` and typed decoding in the future, so I already added those.

Reply:
:+1: If this is code you don't expect to be useful now, but do expect to be useful in the future then that's fine. Just wanted to clarify that as written the `dec_hook` will never be called since no `type=` kwarg was passed in.